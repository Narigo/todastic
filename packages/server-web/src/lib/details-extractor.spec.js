const { extractDetails } = require("./details-extractor");

describe("DetailsExtractor", () => {
  it("extracts labels and tracked times", () => {
    expect(
      extractDetails(
        'wait for this nice #label do you? #TRACK({"trackedTime": "00:20:00", "tracker": "Steve", "date": "2018-09-18 14:06:00"})'
      )
    ).toMatchSnapshot();
  });
  it("returns undefined labels and tracked times", () => {
    // thank you, mongoose
    expect(extractDetails("wait for this nice label do you?")).toMatchSnapshot();
  });
});
