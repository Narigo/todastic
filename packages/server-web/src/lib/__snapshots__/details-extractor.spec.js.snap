// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsExtractor extracts labels and tracked times 1`] = `
Object {
  "labels": Array [
    "#label",
  ],
  "text": "wait for this nice label do you?",
  "trackedTimes": Array [
    Object {
      "date": "2018-09-18 14:06:00",
      "trackedTime": "00:20:00",
      "tracker": "Steve",
    },
  ],
}
`;

exports[`DetailsExtractor returns undefined labels and tracked times 1`] = `
Object {
  "labels": Array [],
  "text": "wait for this nice label do you?",
  "trackedTimes": Array [],
}
`;
